service/ExchangeService.js

```diff
@@ -206,47 +294,125 @@ var fetchFromExmo= function(callback){
 	fetchExchangeData(AppConfig.exchanges.yobit.name, AppConfig.exchanges.yobit.url, function(err, result){
 		if ( err ){
 			callback(err);
+		}else if (result === null) {
+			callback(null, null);
 		}else{
-			var data = {
+			data = {
 				exchange: AppConfig.exchanges.yobit.name,
 				url: AppConfig.exchanges.yobit.orgUrl,
 				price: result.dash_usd.last
 			};
 			data.volume = result.dash_usd.vol*data.price;
-			callback(null, data);
+			fetchExchangeData(AppConfig.exchanges.yobit.name, AppConfig.exchanges.yobit.urlBTC, function(err, result){
+				if ( err ){
+					callback(err);
+				}else if (result === null) {
+					callback(null, null);
+				}else{
+					data['price_btc'] = result.dash_btc.last;
+					fetchExchangeData(AppConfig.exchanges.yobit.name, AppConfig.exchanges.yobit.urlRur, function(err, result){
+						if ( err ){
+							callback(err);
+						}else if (result === null) {
+							callback(null, null);
+						}else{
+							data['price_rur'] = result.dash_rur.last;
+							callback(null, data);
+						}
+					});
+				}
+			});
 		}
 	});
 };
 
 var fetchFromCcex= function(callback){
+	var data = {};
 	log.debug('Fetching exchange data from ' + AppConfig.exchanges.ccex.name + ' at ' + AppConfig.exchanges.ccex.url);
 	fetchExchangeData(AppConfig.exchanges.ccex.name, AppConfig.exchanges.ccex.url, function(err, result){
 		if ( err ){
 			callback(err);
+		}else if (result === null) {
+			callback(null, null);
 		}else{
-			var data = {
+			data = {
 				exchange: AppConfig.exchanges.ccex.name,
 				url: AppConfig.exchanges.ccex.orgUrl,
 				price: result.ticker.lastprice,
 			};
-			callback(null, data);
+			// Fetching BTC price
+			fetchExchangeData(AppConfig.exchanges.ccex.name, AppConfig.exchanges.ccex.urlBTC, function(err, result){
+				if ( err ){
+					callback(err);
+				}else if (result === null) {
+					callback(null, null);
+				}else{
+					data['price_btc'] = result.ticker.lastprice;
+					callback(null, data);
+				}
+			});
 		}
 	});
+
+};
+
+var fetchFromCexio= function(callback){
+	var data = {};
+	log.debug('Fetching exchange data from ' + AppConfig.exchanges.cexio.name + ' at ' + AppConfig.exchanges.cexio.url);
+	fetchExchangeData(AppConfig.exchanges.cexio.name, AppConfig.exchanges.cexio.url, function(err, result){
+		if ( err ){
+			callback(err);
+		}else if (result === null) {
+			callback(null, null);
+		}else{
+			data = {
+				exchange: AppConfig.exchanges.cexio.name,
+				url: AppConfig.exchanges.cexio.orgUrl,
+				price: result.lprice,
+			};
+			fetchExchangeData(AppConfig.exchanges.cexio.name, AppConfig.exchanges.cexio.urlEur, function(err, result){
+				if ( err ){
+					callback(err);
+				}else if (result === null) {
+					callback(null, null);
+				}else{
+					data['price_eur'] = result.lprice;
+					fetchExchangeData(AppConfig.exchanges.cexio.name, AppConfig.exchanges.cexio.urlBTC, function(err, result){
+						if ( err ){
```

Comment:
Can't we use Promises or some async library to avoid such code?

Reply:
Let's do it in the next updates.